cclibs/blocks/promotion/promotion.js

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+// export default async function (block, name, document) {
+export default async function init(el) {
+  const promotionName = el.getAttribute('data-promotion');
+  const response = await window.fetch(`/promotions/hub/${promotionName}.plain.html`);
```

Comment:
I would change the root path here. It will make akamai origin mapping much more complex.

Reply:
You want to avoid having a lot of roots...

```
/hub/
/promotions/
/fragments/
```

etc.

Each time you have a new root, that is more to QA and more work for Akamai.

Reply:
Moved promotions from fedpub into existing creativecloud folder for cc sharepoint